src/Surface.zig

```diff
@@ -540,6 +540,8 @@ pub fn init(
         });
         errdefer io_exec.deinit();
 
+        try self.rt_surface.addSubprocessEnv(&io_exec.subprocess.env);
```

Comment:
I don't like that this traverses 3 completely separate areas of the codebase: core (Surface), termio, and apprt. Further `termio.Exec` isn't meant to be dug into like this, because we may have non-Exec termio implementations in the future (i.e. tmux control protocol).

I suggest we find an alternate approach to this one. Following an approach similar to the cgroup mechanism above is probably _better_, although I don't like that either much and I understand how that's messy with the existing process map.

I don't really have a better solution to consider right now, but just noting this so we don't merge yet.

Reply:
IMHO we already do way too much OS-specific/apprt specific things inside, such as removing GTK-specific env vars that wouldn't be present for non-GTK apprts, removing XCode env vars, handling .hushlogin, etc. It's a huge smorgasbord of platform checks and hacks that already make exposing libghostty as a neutral platform difficult. We should offload all that logic to apprts, even if it does mean there's a three-layer jump in between.

Reply:
Indeed, but there’s a difference between isolated but perhaps domain inconsistent logic and directly accessing fields and decls across components. I’d like to think about a better approach to solve the latter. If it solves the former great but no need to boil the ocean to merge this PR. Just want to avoid the direct coupling.

Reply:
Also just for fun as a point of history, the reason we set platform specific env cars today in termio is because it predates apprts :) it even predates being cross platform and started Linux only!